Dockerfile

```diff
@@ -1,10 +1,23 @@
 FROM kbase/sdkbase2 as build
 
-COPY . /tmp/auth2
-RUN cd /tmp \
-    && git clone https://github.com/kbase/jars \
-    && cd auth2 \
-    && ant buildwar
+WORKDIR /tmp/auth2
+
+# dependencies take a while to D/L, so D/L & cache before the build so code changes don't cause
+# a new D/L
+# can't glob *gradle because of the .gradle dir
+COPY build.gradle gradlew settings.gradle /tmp/auth2/
+COPY gradle/ /tmp/auth2/gradle/
+RUN ./gradlew dependencies
+
+# Now build the code
+COPY deployment/ /tmp/auth2/deployment/
+COPY jettybase/ /tmp/auth2/jettybase/
+COPY src /tmp/auth2/src/
+COPY templates /tmp/auth2/templates/
+COPY war /tmp/auth2/war/
+# for the git commit
+COPY .git /tmp/auth2/.git/
```

Comment:
What's the purpose of moving .git inside the container? To track version?

Reply:
This step needs access to git: https://github.com/kbase/auth2/pull/419/files#diff-49a96e7eea8a94af862798a45174e6ac43eb4f8b4bd40759b5da63ba31ec3ef7R23

It doesn't factor in the final container size due to the 2 stage build